Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/EditorModeMask.shader

```diff
@@ -1,17 +1,29 @@
 {
+    // The editor mode mask shader writes out the visible entity of interest fragments to the red channel and the
+    // visible + occluded fragments to the green channel. In order to achieve this, we turn off depth testing and
+    // instead allow all entity of interest fragments to be rendered, whereupon we perform manual depth testing
+    // in the fragment shader and update the relevent color channels of the mask texture with the closest depth
+    // values of the entities of interest that pass the depth test. As we cannot read and write from and to the
+    // mask texture in the same shader, we instead use the blending stage to accumulate the closest relevent 
+    // fragments described into the mask texture.
+
     "Source" : "EditorModeMask.azsl",
- 
-    "DepthStencilState" : { 
-        "Depth" : { "Enable" : true, "CompareFunc" : "GreaterEqual" }
-    },
 
     "DrawList" : "editormodemask",
 
-    //"RasterState" :
-    //{
-    //    "depthBias" : "10",
-    //    "depthBiasSlopeScale" : "4"
-    //},
+    "DepthStencilState" : 
+    { 
+        "Depth" : { "Enable" : false }
+    },
+
+    // Overwrite the mask texture with any non-zero mask values from the fragment shader output
+    "BlendState" : 
+    {
+        "Enable" : true,
+        "BlendSource" : "AlphaSource",
+        "BlendDest" : "AlphaSourceInverse",
+        "BlendOp" : "Maximum"
+    },
```

Comment:
Very confused about this since the mask pass now uses an R8G8 texture and has no alpha, so this blendstate isn't doing much (unless I'm missing something?)

Reply:
It's a `max` blend op, so acting effectively as a `GreaterEqual` depth test for the "visible fragments only" mask (red channel).

Apologies, the `BlendSrc` and `BlendDst` params where a hangover of a previous experiment and should both now be set to `None` (that's probably what mislead you. Fixed 👍